tox.ini

```diff
@@ -1,67 +1,56 @@
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/c-code
 [tox]
+minversion = 3.18
 envlist =
-    py27,
-    py27-pure,
-    py35,
-    py35-pure,
-    py36,
-    py36-pure,
-    py37,
-    py37-pure,
-    py38,
-    py38-pure,
-    pypy,
-    pypy3,
-    coverage-report,
+    lint
+    py27,py27-pure
+    py35,py35-pure
+    py36,py36-pure
+    py37,py37-pure
+    py38,py38-pure
+    py39,py39-pure
+    pypy
+    pypy3
+    coverage
 
 [testenv]
-install_command = pip install --no-cache-dir --no-binary=:all: {opts} {packages}
-commands =
-    coverage run -m zope.testrunner --test-path={envsitepackagesdir} -s Persistence {posargs:-vc1}
+usedevelop = true
 deps =
-    zope.testrunner
-    coverage
+    wheel
 setenv =
+    pure: PURE_PYTHON=1
+    !pure-!pypy-!pypy3: PURE_PYTHON=0
     PIP_NO_CACHE = 1
-    COVERAGE_FILE=.coverage.{envname}
-
-[testenv:py27-pure]
-basepython =
-    python2.7
-setenv =
-    PURE_PYTHON = 1
-    COVERAGE_FILE=.coverage.{envname}
-
-[testenv:py35-pure]
-basepython =
-    python3.5
-setenv =
-    PURE_PYTHON = 1
-    COVERAGE_FILE=.coverage.{envname}
-
-[testenv:py36-pure]
-basepython =
-    python3.6
-setenv =
-    PURE_PYTHON = 1
-    COVERAGE_FILE=.coverage.{envname}
+commands =
+    zope-testrunner --test-path=src {posargs:-vc}
+extras =
+    test
+install_command = pip install --no-cache-dir --no-binary=:all: {opts} {packages}
```

Comment:
This can be fixed by adding `yum -y install libffi-devel` to the manylinux script.

I'm not 100% sure why that option is there, but I do know that it has been possible to confuse pip into downloading the wrong wheels for the platform in the past.

Reply:
`--no-binary` was added in https://github.com/zopefoundation/Persistence/pull/19, which fixed #18: Apparently, this package didn't (and still doesn't) support `PURE_PYTHON` at runtime, it only checks it when `setup.py` is run. That leads to the possibility of building two wheels with the same name, one of which has the C extension, one of which doesn't. And then if the wrong one gets in the cache…

This package should probably stop doing that.

Reply:
@jamadden I must have totally forgotten that I myself made this change.

Do you think https://github.com/zopefoundation/Persistence/pull/26/commits/6645417c325798c7ed04d271626c3936f8b3750a is enough to fix #27?

Reply:
https://github.com/zopefoundation/Persistence/pull/26/commits/31da5d662847734ea3c7761651afdfbc4e381e8b additionally drops using `--no-binary=:all:`.

Reply:
Yup, I think that should fix the wheel building issue! 

I'm not sure the test is really right, though. It seems to me that it should always be run, and it should assert `CAPI is None if (IS_PYPY or IS_PURE) else CAPI is not None`, followed by the remainder of what it's doing.